azimuth/modules/base_classes/dask_module.py

```diff
@@ -217,6 +217,10 @@ def on_end(self, fut, module):
             self._store_data_in_cache(fut.result(), fut.indices)
             log.info(f"{self.name} completed and stored in cache", status=fut.status)
 
+    def delete_future(self):
```

Comment:
The only question I have is whether the caller (from outside `dask_module.py`) needs to know HOW this works (setting `future` to `None`), or if it would be more user-friendly to simply say WHAT it does (not mentioning that it involves some `future` poutine), for example
```suggestion
    def clear(self):
```

Reply:
I would be ok with it! I'll make the change.

Reply:
Cool! THat way, we keep the implementation for, well, the implementation. 👍